MonoGame.Framework/Graphics/Texture2D.DirectX.cs

```diff
@@ -205,26 +206,36 @@ private int CalculateSubresourceIndex(int arraySlice, int level)
             return arraySlice * _levelCount + level;
         }
 
-        private static Texture2D PlatformFromStream(GraphicsDevice graphicsDevice, Stream stream)
+        private unsafe static Texture2D PlatformFromStream(GraphicsDevice graphicsDevice, Stream stream)
         {
-            if (!stream.CanSeek)
-                throw new NotSupportedException("stream must support seek operations");
+            var reader = new ImageReader();
+            int width, height, channels;
 
-            // For reference this implementation was ultimately found through this post:
-            // http://stackoverflow.com/questions/9602102/loading-textures-with-sharpdx-in-metro 
-            Texture2D toReturn = null;
-            SharpDX.WIC.BitmapDecoder decoder;
+            // The data returned is always four channel BGRA
+            var data = reader.Read(stream, out width, out height, out channels, Imaging.STBI_rgb_alpha);
 
-            using (var bitmap = LoadBitmap(stream, out decoder))
-            using (decoder)
+            // XNA blacks out any pixels with an alpha of zero.
+            if (channels == 4)
             {
-                SharpDX.Direct3D11.Texture2D sharpDxTexture = CreateTex2DFromBitmap(bitmap, graphicsDevice);
+                fixed (byte* b = &data[0])
+                {
+                    for (var i = 0; i < data.Length; i += 4)
+                    {
+                        if (b[i + 3] == 0)
+                        {
+                            b[i + 0] = 0;
+                            b[i + 1] = 0;
+                            b[i + 2] = 0;
+                        }
```

Comment:
Do we know if this is necessary with STB?  We should determine if it internally does this already to avoid the extra work.

Reply:
XNA was unique in this regard by partially premultiplying alpha (only where A = 0).  STB returns the data as in the original PNG.  In my test case, this was RGB = 0xFFFFFF and A = 0.